lib/rspec/rails/matchers/be_a_new.rb

```diff
@@ -49,7 +49,7 @@ def attributes
 
         def attributes_match?(actual)
           attributes.stringify_keys.all? do |key, value|
-            actual.attributes[key].eql?(value)
+            actual.attributes[key].eql?(value) || (value.respond_to?(:expected, true) && actual.attributes[key].include?(value.expected))
```

Comment:
I believe it is covered by tests here:

https://github.com/rspec/rspec-expectations/blob/0ad53b7b0a6e34b1d20f64b1c6588a26648cf6ed/spec/rspec/matchers/built_in/have_attributes_spec.rb#L83-L99

The test coverage there is pretty light because all the heavy lifting is done by other logic the matcher delegates to.